### PR TITLE
second commit with all the code

### PR DESCRIPTION
**Pull request by EitanPinto**

### CODE REVIEW

File: reports.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    DOM.content = document.querySelector("#coinsCardsContent")
+    showOrHideSearchContainer("hidden")
+
+    if (selectedCoins.length) {
+        drawLoader()
+        selectedCoins.forEach((coin) => coin.history = [])
+        await updateChart();
+        setIntervalId = setInterval(updateChart, 2000);
+    } else {
+        DOM.content.innerHTML = "";
```
Comment (galamo) [inline reports.js:83]: you better create a function which lunch+create the sweetAlert content/component

File: reports.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const jsonResult = await result.json();
+
+    const errorMessage = "Error";
+    if (jsonResult.Response === errorMessage) {
+        DOM.content.innerHTML = "";
+        const swalTitleText = "We're Sorry :(";
+        const swalBodyText = "The Selected Coin is Not a Valid coin that corresponds with our Third Party API...You can close this pop-up and return to our Home Page to select a new one";
+        const swalIconType = "error"
+        swal(swalTitleText, swalBodyText, swalIconType, {
+        }).then(() => getCryptoCoinsInit())
```
Comment (galamo) [inline reports.js:51]: you are repeating the same sweetAlert functionality instead creating a function and reuse

File: homePage.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+}
+
+getCryptoCoinsInit()
+
+function handleSearch() {
+    const inputValue = DOM.searchInput.value
+    if (!inputValue) {
+        getCryptoCoinsInit()
+    } else {
+        const filteredCoins = coinsData.filter((coin) => coin.symbol.toLowerCase() === inputValue.toLowerCase())
```
Comment (galamo) [inline homePage.js:72]: inputValue.toLowerCase() can be one line above

and then the loop (filter_) will not need to execute the toLower on every iteration

File: homePage.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    fifthInnerDiv.classList.add("more-info-button-div")
+
+    const MoreInfoButton = document.createElement("button");
+    MoreInfoButton.classList.add("btn", "btn-primary", "btn-design");
+    MoreInfoButton.innerText = "More Info";
+    MoreInfoButton.addEventListener("click", async () => {
+        if (MoreInfoButton.innerText === "More Info") {
+            MoreInfoButton.innerText = "Less Info"
+            forthInnerDiv.style.display = "block"
+            await _getMoreInfo(coinData.id)
```
Comment (galamo) [inline homePage.js:191]: use prefix _ when the method is internal

File: reports.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        await updateChart();
+        setIntervalId = setInterval(updateChart, 2000);
+    } else {
+        DOM.content.innerHTML = "";
+        const swalTitleText = "Please Select at Least 1 Coin to view Chart's Results";
+        const swalBodyText = "You can close this pop-up and return to our Home Page";
+        const swalIconType = "info"
+        swal(swalTitleText, swalBodyText, swalIconType, {
+        }).then(() => getCryptoCoinsInit())
+    }
```
Comment (galamo) [inline reports.js:89]: grade: 98

Comment (EitanPinto) [reply reports.js:89]: noted on all your review points. will fix and update the project.
thank you for this grade !

**Pull request merged**